### PR TITLE
Implement Assessment.QuestionType entities

### PR DESCRIPTION
**Pull request by indocomsoft**
I have not yet written tests for lib/cadet/assessments.ex because @sreycodes is still writing the code for the assessment context.

### CODE REVIEW

File: lib/cadet/assessments/question_types/programming_question.ex [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  alias Cadet.Assessments.QuestionTypes.Library
+
+  embedded_schema do
+    field(:content, :string)
+    field(:solution_template, :string)
+    field(:solution_header, :string)
```
Comment (evansb) [inline lib/cadet/assessments/question_types/programming_question.ex]: This should be optional field, I think it's needed only for path

File: lib/cadet/assessments.ex [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    %MCQQuestion{}
+    |> MCQQuestion.changeset(attr)
+  end
+
+  def create_programmingquestion(json_attr) when is_binary(json_attr) do
+    %ProgrammingQuestion{}
+    |> ProgrammingQuestion.changeset(%{raw_programmingquestion: json_attr})
+  end
+
+  def create_programmingquestion(attr = %{}) do
```
Comment (evansb) [inline lib/cadet/assessments.ex]: `create_programming_question` is a bit easier to read.

File: lib/cadet/assessments.ex [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  alias Cadet.Assessments.QuestionTypes.MCQQuestion
+  alias Cadet.Assessments.QuestionTypes.ProgrammingQuestion
+
+  def create_mcqquestion(json_attr) when is_binary(json_attr) do
+    %MCQQuestion{}
+    |> MCQQuestion.changeset(%{raw_mcqquestion: json_attr})
+  end
+
+  def create_mcqquestion(attr = %{}) do
```
Comment (evansb) [inline lib/cadet/assessments.ex]: `create_multiple_choice_question`

**Comment by indocomsoft** [comment]
Done.

**Pull request merged**